Tensile/KernelWriterAssembly.py

```diff
@@ -6381,8 +6415,16 @@ def globalReadGuardK(self, kernel, tP):
                     vgpr(zeroVgpr), \
                     "vcc", \
                     "gra += 1 (upper)")
+              # pack 2x registers containing 16-bit each into one 32-bit packed half-dword
+              if (kernel["ProblemType"]["DataType"].isHalf() or \
+                kernel["ProblemType"]["DataType"].isBFloat16()) and \
+                kernel["AssertSummationElementMultiple"] % 2 != 0:
+                if r % 2 == 1 and r > 0:
```

Comment:
I don't think it's possible for r to be negative here, was this if meant to test something else?

Reply:
Right. `r>0` shouldn't be here. Probably some code left over from earlier testing